// @prisma/internals/dist/cli/getSchema.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var getSchema_exports = {};
__export(getSchema_exports, {
  createSchemaPathInput: () => import_chunk_2BXDHSSY.createSchemaPathInput,
  getCliProvidedSchemaFile: () => import_chunk_2BXDHSSY.getCliProvidedSchemaFile,
  getConfigProvidedSchemaFile: () => import_chunk_2BXDHSSY.getConfigProvidedSchemaFile,
  getSchemaWithPath: () => import_chunk_2BXDHSSY.getSchemaWithPath,
  getSchemaWithPathOptional: () => import_chunk_2BXDHSSY.getSchemaWithPathOptional,
  printSchemaLoadedMessage: () => import_chunk_2BXDHSSY.printSchemaLoadedMessage
});
module.exports = __toCommonJS(getSchema_exports);
var import_chunk_2BXDHSSY = require("../chunk-2BXDHSSY.js");
var import_chunk_PG5FDKSF = require("../chunk-PG5FDKSF.js");
var import_chunk_4VNS5WPM = require("../chunk-4VNS5WPM.js");
