// @clerk/shared/dist/runtime/telemetry.js
require('./_chunks/constants-BkJLCu9f.js');
require('./_chunks/isomorphicAtob-BotUVOm6.js');
require('./_chunks/isomorphicBtoa-o8eXb3FF.js');
require('./_chunks/keys-COv7Q4fb.js');
require('./_chunks/getEnvVariable-HbsW2WWf.js');
require('./_chunks/runtimeEnvironment-DLM4o1Pu.js');
require('./_chunks/underscore-CWzKsZlj.js');
const require_telemetry = require('./_chunks/telemetry-uUa2d_fu.js');

exports.EVENT_SAMPLING_RATE = require_telemetry.EVENT_SAMPLING_RATE;
exports.EVENT_THEME_USAGE = require_telemetry.EVENT_THEME_USAGE;
exports.TelemetryCollector = require_telemetry.TelemetryCollector;
exports.eventComponentMounted = require_telemetry.eventComponentMounted;
exports.eventFrameworkMetadata = require_telemetry.eventFrameworkMetadata;
exports.eventMethodCalled = require_telemetry.eventMethodCalled;
exports.eventPrebuiltComponentMounted = require_telemetry.eventPrebuiltComponentMounted;
exports.eventPrebuiltComponentOpened = require_telemetry.eventPrebuiltComponentOpened;
exports.eventThemeUsage = require_telemetry.eventThemeUsage;
