// @clerk/shared/dist/runtime/utils/index.js
require('../_chunks/getEnvVariable-HbsW2WWf.js');
const require_runtimeEnvironment = require('../_chunks/runtimeEnvironment-DLM4o1Pu.js');
const require_handleValueOrFn = require('../_chunks/handleValueOrFn-D1JzdrJ-.js');
const require_instance = require('../_chunks/instance-DPIKVAEY.js');
const require_noop = require('../_chunks/noop-CTUdRHpt.js');
const require_createDeferredPromise = require('../_chunks/createDeferredPromise-B0t_CzEQ.js');
const require_utils = require('../_chunks/utils-COuXXlVE.js');

exports.allSettled = require_utils.allSettled;
exports.automatedEnvironmentVariables = require_runtimeEnvironment.automatedEnvironmentVariables;
exports.createDeferredPromise = require_createDeferredPromise.createDeferredPromise;
exports.fastDeepMergeAndKeep = require_utils.fastDeepMergeAndKeep;
exports.fastDeepMergeAndReplace = require_utils.fastDeepMergeAndReplace;
exports.handleValueOrFn = require_handleValueOrFn.handleValueOrFn;
exports.isAutomatedEnvironment = require_runtimeEnvironment.isAutomatedEnvironment;
exports.isDevelopmentEnvironment = require_runtimeEnvironment.isDevelopmentEnvironment;
exports.isProductionEnvironment = require_runtimeEnvironment.isProductionEnvironment;
exports.isStaging = require_instance.isStaging;
exports.isTestEnvironment = require_runtimeEnvironment.isTestEnvironment;
exports.logErrorInDevMode = require_utils.logErrorInDevMode;
exports.noop = require_noop.noop;
exports.runIfFunctionOrReturn = require_utils.runIfFunctionOrReturn;
